test/unit/3p/test-3p-frame.js

```diff
@@ -34,623 +35,643 @@ import {
-
-      const div = document.createElement('my-element');
-      div.setAttribute('data-test-attr', 'value');
-      div.setAttribute('data-ping', 'pong');
-      div.setAttribute('width', '50');
-      div.setAttribute('height', '100');
-
-      const {innerWidth: width, innerHeight: height} = window;
-      setupElementFunctions(div);
-
-      const viewer = Services.viewerForDoc(window.document);
-      const viewerMock = window.sandbox.mock(viewer);
-      viewerMock
-        .expects('getUnconfirmedReferrerUrl')
-        .returns('http://acme.org/')
-        .once();
-
-      container.appendChild(div);
-
-      window.sandbox
-        .stub(DomFingerprint, 'generate')
-        .callsFake(() => 'MY-MOCK-FINGERPRINT');
-
-      const iframe = getIframe(window, div, '_ping_', {clientId: 'cidValue'});
-      const {src} = iframe;
-      const locationHref = location.href;
-      expect(locationHref).to.not.be.empty;
-      const docInfo = Services.documentInfoForDoc(window.document);
-      expect(docInfo.pageViewId).to.not.be.empty;
-      const name = JSON.parse(iframe.name);
-      const sentinel = name.attributes._context['sentinel'];
-      const fragment = {
-        'testAttr': 'value',
-        'ping': 'pong',
-        'width': 50,
-        'height': 100,
-        'type': '_ping_',
-        '_context': {
-          'referrer': 'http://acme.org/',
-          'ampcontextVersion': '$internalRuntimeVersion$',
-          'ampcontextFilepath':
-            'https://3p.ampproject.net/$internalRuntimeVersion$/ampcontext-v0.js',
-          'canonicalUrl': docInfo.canonicalUrl,
-          'sourceUrl': locationHref,
-          'pageViewId': docInfo.pageViewId,
-          'clientId': 'cidValue',
-          'initialLayoutRect': div.getPageLayoutBox(),
-          'location': {'href': locationHref},
-          'tagName': 'MY-ELEMENT',
-          'mode': {
-            'localDev': true,
-            'development': false,
-            'minified': false,
-            'test': false,
-            'version': '$internalRuntimeVersion$',
-            'esm': false,
-          },
-          'canary': true,
-          'hidden': false,
-          // Note that DOM fingerprint will change if the document DOM changes
-          // Note also that running it using --files uses different DOM.
-          'domFingerprint': '1725030182',
-          'startTime': 1234567888,
-          'experimentToggles': {'exp-a': true, 'exp-b': true},
-          'sentinel': sentinel,
-          'initialIntersection': {
-            'time': 1234567888,
-            'rootBounds': {
-              'left': 0,
-              'top': 0,
-              'width': width,
-              'height': height,
-              'bottom': height,
-              'right': width,
-              'x': 0,
-              'y': 0,
+      // TODO(bradfrizzell) break this out into a test-iframe-attributes
+      it('should create an iframe', () => {
+        window.AMP_CONFIG = {canary: true};
```

Comment:
why canary here?

Reply:
The top window gets canary in its config in our tests. But it's not propagated to child windows, so I figured I'll just set it here so that all tests get the right value. A smaller diff.